src/rgw/rgw_sal.h

```diff
@@ -314,6 +315,10 @@ class Store {
     virtual int forward_request_to_master(const DoutPrefixProvider *dpp, User* user, obj_version* objv,
 					  bufferlist& in_data, JSONParser* jp, req_info& info,
 					  optional_yield y) = 0;
+    virtual int forward_request_to_master(const DoutPrefixProvider *dpp, const RGWAccessKey& key, obj_version* objv,
+					     bufferlist& in_data,
+					     RGWXMLDecoder::XMLParser* parser, req_info& info,
+					     optional_yield y) = 0;
```

Comment:
can we please find a way to avoid needing two overloads of `forward_request_to_master()`?

Reply:
Yes, sure.

Reply:
done.